### PR TITLE
Properly handle passing (nested) Dictionaries/Structure containing arrays

### PR DESCRIPTION
**Pull request by josephcoombe**

### CODE REVIEW

**Comment by josephcoombe** [comment]
We still don't handle cases where a user might pass an array of structs (/ list of dictionaries)

**Comment by josephcoombe** [comment]
https://www.mathworks.com/help/matlab/matlab_external/handle-data-returned-from-matlab-to-python.html

> The MATLAB Engine API for Python does not support these MATLAB data types.
> ...
> - Structure array

File: matlab_proxy/matlab_proxy/__init__.py [hunk 1]
```diff
@@ -108,6 +108,20 @@ def invoke(self, name, args, nargout, bare, input_names, output_names):
...
         args = pickle.loads(b64decode(args))
+
+        def convert_arg_to_matlab_datatype(arg):
+            if isinstance(arg, list):
+                if len(arg) and isinstance(arg[0], six.string_types):
+                    return arg
+                else:
+                    return matlab.double(arg)
+            elif isinstance(arg, dict):
+                for k, v in arg.iteritems():
```
Comment (ksmyth) [inline matlab_proxy/matlab_proxy/__init__.py:119]: This is not Python 3 compatible

Comment (ksmyth) [reply matlab_proxy/matlab_proxy/__init__.py:119]: https://six.readthedocs.io/#six.iteritems

Comment (josephcoombe) [reply matlab_proxy/matlab_proxy/__init__.py:119]: Corrected
https://github.com/metamorph-inc/matlab_wrapper/commit/d207a403625c5c6734f662466878febb50795273

**Pull request merged**